### PR TITLE
merge configSensi with param sensi analysis

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/components/dialogs/parameters/sensi/sensi-hvdcs.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
+/**
```
Comment (trikimahdi) [inline src/components/dialogs/parameters/sensi/sensi-hvdcs.js:1]: This is a new file. It has to be on Typescript

File: src/components/dialogs/parameters/sensi/sensi-hvdcs.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    };
+};
+const SensiHVDCs = () => {
+    const intl = useIntl();
+
+    const useFieldArrayOutput = useFieldArray({
+        name: `${PARAMETER_SENSI_HVDC}`,
+    });
+
+    const COLUMNS_DEFINITIONS = useMemo(() => {
```
Comment (trikimahdi) [inline src/components/dialogs/parameters/sensi/sensi-hvdcs.js:100]: create a new typescript file to define this

File: src/components/dialogs/parameters/sensi/sensi-injections-set.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
+/**
```
Comment (trikimahdi) [inline src/components/dialogs/parameters/sensi/sensi-injections-set.js:1]: It is almost the same code as the previous file (sensi-hvdcs.js). Can we refactor these 2 files ?

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                                            value={subIndex}
+                                            label={
+                                                <FormattedMessage
+                                                    id={subTab.label}
+                                                />
+                                            }
+                                        ></Tab>
+                                    ))}
+                                </Tabs>
+                                <TabPanel
```
Comment (trikimahdi) [inline src/components/dialogs/parameters/sensi/sensi-parameters-selector.js:135]: We can use a map to list all TabPanels

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+import { EQUIPMENT_TYPES } from '../../../utils/equipment-types';
+import { Tab, Tabs } from '@mui/material';
+import { TabPanel } from '../parameters';
+import SensiInjectionsSet from './sensi-injections-set';
+import SensiInjection from './sensi-injections';
+import SensiHVDCs from './sensi-hvdcs';
+import SensiPSTs from './sensi-psts';
+import SensiNodes from './sensi-nodes';
+
+export const MONITORED_BRANCHES_EQUIPMENT_TYPES = [
```
Comment (trikimahdi) [inline src/components/dialogs/parameters/sensi/sensi-parameters-selector.js:23]: I think these declarations have to be on a separate file

File: src/components/dialogs/parameters/sensi/sensitivity-Flow-parameters.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    FLOW_VOLTAGE_SENSITIVITY_VALUE_THRESHOLD,
+} from '../../../utils/field-constants';
+import { FormattedMessage } from 'react-intl';
+import { useStyles } from '../parameters';
+import { FloatInput } from '@gridsuite/commons-ui';
+
+const SensitivityAnalysisFields = () => {
+    const classes = useStyles();
+
+    const flowFlowSensitivityValueThresholdField = (
```
Comment (trikimahdi) [inline src/components/dialogs/parameters/sensi/sensitivity-Flow-parameters.js:23]: We can make an array that contains FloatInputNames and make a map of this array to list FloatInput. The 3 FloatInput are the same except the name.

File: src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+    getSensiPSTsFormSchema,
+} from './sensi-psts';
+import {
+    getSensiNodesformatNewParams,
+    getSensiNodesFormSchema,
+} from './sensi-nodes';
+import SensiParametersSelector from './sensi-parameters-selector';
+import { LineSeparator } from '../../dialogUtils';
+
+export const useGetSensitivityAnalysisParameters = () => {
```
Comment (trikimahdi) [inline src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js:68]: This custom hook have to be on a seperate file. There is a file named custom-hooks. We can use it

**Review by trikimahdi — CHANGES_REQUESTED** [review]

File: src/components/dialogs/parameters/sensi/sensi-hvdcs.js (hunk 1 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-hvdcs.js]: done

File: src/components/dialogs/parameters/sensi/sensi-hvdcs.js (hunk 2 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-hvdcs.js]: done

File: src/components/dialogs/parameters/sensi/sensi-injections-set.js (hunk 3 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-injections-set.js]: done

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js (hunk 5 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-parameters-selector.js]: done

File: src/components/dialogs/parameters/sensi/sensitivity-Flow-parameters.js (hunk 6 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensitivity-Flow-parameters.js]: done

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js (hunk 4 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-parameters-selector.js:119]: done

File: src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js [hunk 8]
```diff
@@ -0,0 +1,433 @@
...
+                    <SubmitButton onClick={handleSubmit(onSubmit)}>
+                        <FormattedMessage id="validate" />
+                    </SubmitButton>
+                    <CloseButton
+                        hideParameters={hideParameters}
+                        className={classes.button}
+                    />
+                </DialogActions>
+            </Grid>
+        </FormProvider>
```
Comment (Tristan-WorkGH) [inline src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js:431]: the unifying of the visual of parameters in #1423 & #1520 isn't respected
```suggestion
        <FormProvider validationSchema={formSchema} {...formMethods}>
            <Grid
                container
                spacing={1}
                padding={1}
                key="sensitivityAnalysisParameters"
            >
                <Grid
                    container
                    spacing={1}
                    sx={{ padding: 0, paddingBottom: 1 }}
                >
                    <DropDown
                        value={provider}
                        label="Provider"
                        values={providers}
                        callback={updateProviderCallback}
                    />
                </Grid>
                <LineSeparator />
                <SensitivityAnalysisFields
                    reset={reset}
                    useSensitivityAnalysisParameters={
                        useSensitivityAnalysisParameters
                    }
                />
                <LineSeparator />
                <SensiParametersSelector
                    reset={reset}
                    useSensitivityAnalysisParameters={
                        useSensitivityAnalysisParameters
                    }
                />
            </Grid>
            <Grid
                container
                className={classes.controlItem + ' ' + classes.marginTopButton}
                maxWidth="md"
            >
                <LabelledButton
                    callback={clear}
                    label="resetToDefault"
                />
                <SubmitButton onClick={handleSubmit(onSubmit)}>
                    <FormattedMessage id="validate" />
                </SubmitButton>
                <CloseButton
                    hideParameters={hideParameters}
                    className={classes.button}
                />
            </Grid>
        </FormProvider>
```

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: decs.d.ts [hunk 9]
```diff
@@ -135,4 +135,11 @@ declare module '@gridsuite/commons-ui' {
     }
 
     export const FlatParameters: FunctionComponent<FlatParametersProps>;
+
+    export const elementType = {
```
Comment (flomillot) [inline decs.d.ts:139]: Why do you put this here ?

Comment (flomillot) [reply decs.d.ts:139]: This file is to define TypeScript interfaces when we import JavaScript libraries (commonly commons-ui)

File: src/translations/fr.json [hunk 10]
```diff
@@ -938,11 +939,15 @@
 
     "resultsThreshold": "Seuil minimal de sensibilité",
     "SensitivityBranches": "Sensibilité au niveau des quadripôles",
+    "SensitivityNodes": "Sensibilité au niveau des nœuds",
+    "SensiInjectionsSet": "Sensi Injections set",
+    "SensiInjection": "Sensi Injections",
+    "SensiHVDC": "Sensi Hvdc",
+    "SensiPST": "Sensi PST",
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/translations/fr.json]: done

File: src/translations/fr.json [hunk 11]
```diff
@@ -938,11 +939,15 @@
...
     "SensitivityInjectionsSet": "Par rapport à un ensemble d'injections :",
     "SensitivityInjection": "Par rapport à chaque injection :",
-    "SensitivityHvdc": "Par rapport à chaque HVDC :",
-    "SensitivityTD": "Par rapport à chaque TD :",
-    "SensitivityNodes": "Sensibilité au niveau des nœuds",
+    "SensitivityHVDC": "Par rapport à chaque HVDC :",
+    "SensitivityPST": "Par rapport à chaque TD :",
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/translations/fr.json]: done

File: src/translations/fr.json [hunk 12]
```diff
@@ -954,7 +959,7 @@
     "AddSensitivityFactor": "Ajouter calcul de sensibilité",
     "ContingencyLists": "Aléas",
     "Injections": "Injections",
-    "SupervisedVoltageLevels": "Postes surveillés",
+    "MonitoredVoltageLevels": "Postes surveillés",
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/translations/fr.json:962]: done

File: src/translations/fr.json [hunk 13]
```diff
@@ -938,11 +939,15 @@
...
+    "SensitivityNodes": "Sensibilité au niveau des nœuds",
+    "SensiInjectionsSet": "Sensi Injections set",
+    "SensiInjection": "Sensi Injections",
+    "SensiHVDC": "Sensi Hvdc",
+    "SensiPST": "Sensi PST",
     "SensitivityInjectionsSet": "Par rapport à un ensemble d'injections :",
     "SensitivityInjection": "Par rapport à chaque injection :",
-    "SensitivityHvdc": "Par rapport à chaque HVDC :",
-    "SensitivityTD": "Par rapport à chaque TD :",
-    "SensitivityNodes": "Sensibilité au niveau des nœuds",
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/translations/fr.json:945]: done

File: src/services/study/sensitivity-analysis.js [hunk 14]
```diff
@@ -12,31 +12,16 @@ import {
...
-
-    return backendFetch(url, {
-        method: 'post',
-        headers: {
-            Accept: 'application/json',
-            'Content-Type': 'application/json',
-        },
-        body: body,
-    });
+    console.debug(startSensiAnalysisUrl);
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/services/study/sensitivity-analysis.js:23]: yes

File: src/hooks/use-create-row-data-sensi.ts [hunk 15]
```diff
@@ -0,0 +1,21 @@
+import { SELECTED } from '../components/utils/field-constants';
+import { useFieldArray } from 'react-hook-form';
+import { ISensiParameters } from '../components/dialogs/parameters/sensi/columns-definitions';
+import { useMemo } from 'react';
+
+export const useCreateRowDataSensi = (sensiParam: ISensiParameters) => {
+    const useFieldArrayOutput = useFieldArray({
+        name: `${sensiParam.name}`,
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/hooks/use-create-row-data-sensi.ts]: done

File: src/components/utils/dnd-table/dnd-table.js [hunk 16]
```diff
@@ -132,6 +135,13 @@ function EditableTableCell({
                     hideErrorMessage={true}
                 />
             )}
+            {column.menuItem && (
+                <SelectInput
+                    name={`${arrayFormName}[${rowIndex}].${column.dataKey}`}
+                    options={column.equipmentTypes}
+                    size={'small'}
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/components/utils/dnd-table/dnd-table.js]: done

File: src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js (hunk 8 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensitivity-analysis-parameters.js]: done

File: src/translations/fr.json [hunk 17]
```diff
@@ -939,9 +940,13 @@
...
     "SensitivityInjection": "Par rapport à chaque injection :",
-    "SensitivityHvdc": "Par rapport à chaque HVDC :",
-    "SensitivityTD": "Par rapport à chaque TD :",
+    "SensitivityHVDC": "Par rapport à chaque HVDC :",
+    "SensitivityPST": "Par rapport à chaque TD :",
     "SensitivityNodes": "Sensibilité au niveau des nœuds",
+    "SensiInjectionsSet": "Sensi Injections set",
+    "SensiInjection": "Sensi Injections",
+    "SensiHVDC": "Sensi Hvdc",
+    "SensiPST": "Sensi PST",
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/translations/fr.json:949]: sensitivityInjectionsSet ( parameters ) 
SensiInjectionsSet ( arrayFormName )

is the same for the others

**Review by Tristan-WorkGH — APPROVED** [review]

**Review by EstherDarkish — CHANGES_REQUESTED** [review]
- There is no scrolling bar in the sensitivity analysis parameter form, so we cannot validate the form in small screen.
- The buttons on the bottom of the form are aligned on the left instead of the right.
- The tabs' alignments are not correct, they should be on the same level, and a little more on the left. There seems to be a margin or padding messing with it, coming from the parent's style.

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js [hunk 18]
```diff
@@ -0,0 +1,229 @@
...
+    );
+
+    return (
+        <>
+            <Grid item maxWidth="md" width="100%">
+                <Tabs
+                    value={tabValue}
+                    variant="scrollable"
+                    onChange={handleTabChange}
+                    //classes={getTabIndicatorClass(tabValue)}
```
Comment (Walid-Sahnoun) [inline src/components/dialogs/parameters/sensi/sensi-parameters-selector.js:101]: remove commented code

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+                >
+                    {tabInfo.map((tab, index) => (
+                        <Tab
+                            key={tab.label}
+                            label={<FormattedMessage id={tab.label} />}
+                            value={index}
+                            //className={getTabClass(index)}
```
Comment (Walid-Sahnoun) [inline src/components/dialogs/parameters/sensi/sensi-parameters-selector.js:108]: also here

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+                {tabInfo.map((tab, index) => (
+                    <TabPanel key={tab.label} value={tabValue} index={index}>
+                        {tabValue === TAB_VALUES.SensitivityBranches &&
+                        tab.subTabs ? (
+                            <>
+                                <Tabs
+                                    value={subTabValue}
+                                    variant="scrollable"
+                                    onChange={handleSubTabChange}
+                                    //classes={getTabIndicatorClass(subTabValue)}
```
Comment (Walid-Sahnoun) [inline src/components/dialogs/parameters/sensi/sensi-parameters-selector.js:122]: commented code

File: src/components/dialogs/parameters/parameters.js [hunk 21]
```diff
@@ -633,18 +634,19 @@ const Parameters = ({ user, isParametersOpen, hideParameters }) => {
         setSecurityAnalysisParameters
     );
 
-    const sensitivityAnalysisParametersBackend = useParametersBackend(
+    const sensibilityAnalysisBackend = useParametersBackend(
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/components/dialogs/parameters/parameters.js]: sensitivity
done

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js (hunk 18 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-parameters-selector.js]: done

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js (hunk 19 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-parameters-selector.js]: done

File: src/components/dialogs/parameters/sensi/sensi-parameters-selector.js (hunk 20 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensi-parameters-selector.js]: done

File: src/translations/fr.json [hunk 22]
```diff
@@ -953,7 +958,7 @@
     "AddSensitivityFactor": "Ajouter calcul de sensibilité",
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/translations/fr.json]: done

File: src/components/dialogs/sensi/sensi-parameters-selector.js [hunk 23]
```diff
@@ -1,314 +0,0 @@
-/**
- * Copyright (c) 2022, RTE (http://www.rte-france.com)
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
```
Comment (EstherDarkish) [inline src/components/dialogs/sensi/sensi-parameters-selector.js:7]: Delete this file

File: src/hooks/use-create-row-data-sensi.ts [hunk 24]
```diff
@@ -0,0 +1,21 @@
+import { SELECTED } from '../components/utils/field-constants';
```
Comment (Walid-Sahnoun) [inline src/hooks/use-create-row-data-sensi.ts:1]: add copyright

**Review by trikimahdi — APPROVED** [review]

File: src/hooks/use-create-row-data-sensi.ts (hunk 24 above)
Comment (ghazwarhili) [reply src/hooks/use-create-row-data-sensi.ts:8]: done

**Review by etiennehomer — CHANGES_REQUESTED** [review]
You have errors in the console to be fixed

**Review by EstherDarkish — COMMENTED** [review]
Some remarks after testing : 
- In the tables, there is the option to drag and drop lines, meaning the line's order is important, yet it can change sometimes.
- The save/validate system is inconsistent : the rows are not taken into account if their checkboxes are not checked, yet if we change an already saved row's Distribution Type, the value is submited when we click Validate and the checkbox is not checked. Or is it because now the checkbox is not needed anymore ?
- The row deletion seems to not be working correctly between two tabs of the same study

**Comment by ghazwarhili** [comment]
> Some remarks after testing :
> 
>     * In the tables, there is the option to drag and drop lines, meaning the line's order is important, yet it can change sometimes.
> 
>     * The save/validate system is inconsistent : the rows are not taken into account if their checkboxes are not checked, yet if we change an already saved row's Distribution Type, the value is submited when we click Validate and the checkbox is not checked. Or is it because now the checkbox is not needed anymore ?
> 
>     * The row deletion seems to not be working correctly between two tabs of the same study

    In the tables, there is the option to drag and drop lines, meaning the line's order is important, yet it can change sometimes.
    => The format (dragAndDrop, selected, delete, the two arrows, etc.) has been approved by the PO, but I can ask them again to confirm.

    The save/validate system is inconsistent: the rows are not taken into account if their checkboxes are not checked. However, if we change the Distribution Type of an already saved row, the value is submitted when we click Validate, and the checkbox is not checked. Or is it because the checkbox is no longer needed?
    => This is intentional (we no longer need the selected checkbox; instead, we added a new column, 'Active,' to enable or disable the configuration line in the back side).

    The row deletion seems to not be working correctly between two tabs of the same study.
    => I just tested line deletion for the same study but opened in two tabs, and it works.
Could you please provide more details about the scenario you followed ?

**Review by EstherDarkish — APPROVED** [review]
Works, but we will have to fix some minor displaying issues in a future user story.

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**